### PR TITLE
codeql workflow for static analysis

### PR DESCRIPTION
**Pull request by andrewc12**
ala https://github.com/openzfs/zfs/pull/14087

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Performance enhancement (non-breaking change which improves efficiency)
- [ ] Code cleanup (non-breaking change which makes code smaller or more readable)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Library ABI change (libzfs, libzfs\_core, libnvpair, libuutil and libzfsbootenv)
- [ ] Documentation (a change to man pages or other documentation)

### Checklist:

- [ ] My code follows the OpenZFS [code style requirements](https://github.com/openzfs/zfs/blob/master/.github/CONTRIBUTING.md#coding-conventions).
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [**contributing** document](https://github.com/openzfs/zfs/blob/master/.github/CONTRIBUTING.md).
- [ ] I have added [tests](https://github.com/openzfs/zfs/tree/master/tests) to cover my changes.
- [ ] I have run the ZFS Test Suite with this change applied.
- [ ] All commit messages are properly formatted and contain [`Signed-off-by`](https://github.com/openzfs/zfs/blob/master/.github/CONTRIBUTING.md#signed-off-by).

### CODE REVIEW

**Review by ryao — CHANGES_REQUESTED** [review]
This should be changed to run on PRs.

**Comment by andrewc12** [comment]
> This should be changed to run on PRs.

Yeah it should. Thanks for finding this.

I also need to change the other workflow I based it on

**Comment by ryao** [comment]
> > This should be changed to run on PRs.
> 
> Yeah it should. Thanks for finding this.
> 
> I also need to change the other workflow I based it on

Look at the PR for the main repository. It shows how to make this run on PRs. :)

**Review by ryao — APPROVED** [review]
It looks good to me, although you probably want to ask for feedback from @gmelikov. He is the one most experienced with github workflows. My knowledge on the topic is shallow compared to his.

File: .github/workflows/codeql-macos.yaml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+      - name: autogen
+        run: |
+          ./autogen.sh
+      - name: configure
+        run: |
+          #https://stackoverflow.com/a/62591864
+          ./configure CPPFLAGS="-I/usr/local/opt/gettext/include -I/usr/local/opt/openssl@1.1/include" LDFLAGS="-L/usr/local/opt/gettext/lib/ -L/usr/local/opt/openssl@1.1/lib"
+      - name: build
+        run: |
+          make -j 2
```
Comment (gmelikov) [inline .github/workflows/codeql-macos.yaml]: You may want to set it dinamically, for Linux it's `-j$(nproc)`, google tells me it's `-j$(sysctl -n hw.ncpu)` for macos

**Review by gmelikov — APPROVED** [review]
LGTM code-wise, I won't challenge you on styles (unneeded commented parts). You may want to change repos' settings to make warnings fatal https://docs.github.com/en/code-security/code-scanning/automatically-scanning-your-code-for-vulnerabilities-and-errors/configuring-code-scanning#defining-the-severities-causing-pull-request-check-failure

**Comment by andrewc12** [comment]
The codeQL scan for macos has 43 alerts compared to upstream which I think was 26
chrome_screenshot_1667007927306

**Comment by andrewc12** [comment]
> LGTM code-wise, I won't challenge you on styles (unneeded commented parts). You may want to change repos' settings to make warnings fatal.

While this is a good idea, I don't know if existing alerts cause PR failures

**Pull request merged**